File: backend/AssignmentSubmissionSystem.API/Models/TeacherAssignment.cs
namespace AssignmentSubmissionSystem.API.Models;

public class TeacherAssignment
{
    public int Id { get; set; }

    public int TeacherId { get; set; }
    public Teacher Teacher { get; set; } = null!;

    public int AcademicClassId { get; set; }
    public AcademicClass AcademicClass { get; set; } = null!;

    public int SubjectId { get; set; }
    public Subject Subject { get; set; } = null!;
}

File: frontend/assignmentsubmissionsystem.web/lib/validations/teacherAssignment.ts
import { z } from "zod";

export const createTeacherAssignmentSchema = z.object({
  teacherId: z.number().int().positive("Please select a teacher."),

  academicClassId: z
    .number()
    .int()
    .positive("Please select an academic class."),

  subjectId: z.number().int().positive("Please select a subject."),
});

export const updateTeacherAssignmentSchema = z.object({
  teacherId: z.number().int().positive("Please select a teacher."),

  academicClassId: z
    .number()
    .int()
    .positive("Please select an academic class."),

  subjectId: z.number().int().positive("Please select a subject."),
});

export type CreateTeacherAssignmentForm = z.infer<
  typeof createTeacherAssignmentSchema
>;

export type UpdateTeacherAssignmentForm = z.infer<
  typeof updateTeacherAssignmentSchema
>;
